### PR TITLE
fix: reset validator signing info missed blocks counter

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

In v24, we upgraded to the sdk v0.50 slashing logic. When timing the migration, we realized that a non-negligible period of time was spent copying over the missed blocks, so we decided it was fine to clear these missed blocks. We, however, forgot to also reset the missed block counter. Therefore, a validator's missed block counter floor is stuck at the value it was at the time of upgrade.

## Testing and Verifying

Go test added to demonstrate the counter is reset.

## Summary by CodeRabbit

- **New Features**
  - Introduced functionality to reset validator signing info missed blocks counter.
  - Implemented operations speedup and various optimizations.
- **Bug Fixes**
  - Removed legacy behavior preventing unbonding lockups at very small block heights on testnets.
- **Tests**
  - Added test cases for new upgrade functionalities in the slashing module.
- **Documentation**
  - Updated CHANGELOG.md with recent changes and improvements.

### CODE REVIEW

**Review by PaddyMc — APPROVED** [review]

File: app/upgrades/v25/upgrades.go [hunk 1]
```diff
@@ -23,8 +26,25 @@ func CreateUpgradeHandler(
...
 }
+
+// resetMissedBlocksCounter resets the missed blocks counter for all validators back to zero.
+// This corrects a mistake that was overlooked in v24, where we cleared all missedBlocks but did not reset the counter.
+func resetMissedBlocksCounter(ctx sdk.Context, slashingKeeper *slashing.Keeper) {
+	// Iterate over all validators signing info
+	slashingKeeper.IterateValidatorSigningInfos(ctx, func(address sdk.ConsAddress, info slashingtypes.ValidatorSigningInfo) (stop bool) {
+		// Reset missed blocks counter
+		info.MissedBlocksCounter = 0
+		slashingKeeper.SetValidatorSigningInfo(ctx, address, info)
```
[the remark this answers is not in the document]
Comment (czarcas7ic) [reply app/upgrades/v25/upgrades.go:46]: Oh I see, couldn't we just reset the bit array to zero again for simplicity @ValarDragon

[the remark this answers is not in the document]
Comment (czarcas7ic) [reply app/upgrades/v25/upgrades.go:46]: By bit array to zero I mean, just deleting them and start from where we did in the migration

[the remark this answers is not in the document]
Comment (ValarDragon) [reply app/upgrades/v25/upgrades.go:46]: Yes that would work!

[the remark this answers is not in the document]
Comment (czarcas7ic) [reply app/upgrades/v25/upgrades.go:46]: Good catch btw, my bad for overlooking

[the remark this answers is not in the document]
Comment (czarcas7ic) [reply app/upgrades/v25/upgrades.go]: I believe I did it correctly here, and added a test that replicates mainnet better: https://github.com/osmosis-labs/osmosis/pull/8053/commits/fbcd43cac63d921d618c35a16438562a2d4c964b

**Review by ValarDragon — APPROVED** [review]
Lgtm+

**Review by PaddyMc — APPROVED** [review]
GG!

**Pull request merged**